### PR TITLE
Multitarget VBCSCompiler.csproj

### PR DESCRIPTION
**Pull request by khyperia**
This is part of work to make the compiler server xplat.

Currently, this change is only targeted at getting VBCSCompiler on desktop/coreclr, but both on Windows (not coreclr+mac/linux).

There will be a follow-up PR to make VBCSCompilerTests multitargeted (but that requires a larger set of changes - I'm currently working on nuking CscCore and making csc multitargeted, which is required (or at least makes it significantly easier))

cc @jaredpar

### CODE REVIEW

File: src/Compilers/Extension/CompilerExtension.csproj [hunk 1]
```diff
@@ -102,6 +102,7 @@
       <IncludeOutputGroupsInVSIX>BuiltProjectOutputGroup</IncludeOutputGroupsInVSIX>
       <IncludeOutputGroupsInVSIXLocalOnly>DebugSymbolsProjectOutputGroup%3b</IncludeOutputGroupsInVSIXLocalOnly>
       <ForceIncludeInVSIX>true</ForceIncludeInVSIX>
+      <AdditionalProperties>TargetFramework=net46</AdditionalProperties>
```
Comment (jaredpar) [inline src/Compilers/Extension/CompilerExtension.csproj]: please add a tracking bug

File: src/Compilers/Server/VBCSCompiler/DesktopBuildServerController.cs [hunk 2]
```diff
@@ -34,38 +28,35 @@ protected override IClientConnectionHost CreateClientConnectionHost(string pipeN
...
+                    // This is a one time server entry.
+                    return null;
                 }
                 else
                 {
-                    return ServerDispatcher.DefaultServerKeepAlive;
+                    return TimeSpan.FromSeconds(keepAliveValue);
                 }
             }
-            catch (ConfigurationErrorsException e)
```
Comment (jaredpar) [inline src/Compilers/Server/VBCSCompiler/DesktopBuildServerController.cs]: >catch (ConfigurationErrorsException e) [](start = 12, length = 38)

this needs to be kept. important case when the config file is corrupt

Comment (jaredpar) [reply src/Compilers/Server/VBCSCompiler/DesktopBuildServerController.cs]: can just catch exception if neccessary

---
In reply to: [138463943](https://github.com/dotnet/roslyn/pull/22075#discussion_r138463943) [](ancestors = 138463943)

Comment (khyperia) [reply src/Compilers/Server/VBCSCompiler/DesktopBuildServerController.cs]: Wait. So the *indexer* on `NameValueCollection` throws on corruption, not when loading the config into `NameValueCollection` via `System.Configuration.ConfigurationManager.AppSettings`?

File: src/Compilers/Server/VBCSCompiler/NamedPipeClientConnection.cs [hunk 3]
```diff
@@ -148,9 +149,19 @@ private NamedPipeServerStream ConstructPipe(string pipeName)
                 PipeTransmissionMode.Byte,
                 PipeOptions.Asynchronous | PipeOptions.WriteThrough,
                 PipeBufferSize, // Default input buffer
-                PipeBufferSize // Default output buffer
-            );
-            pipeStream.SetAccessControl(security);
+                PipeBufferSize, // Default output buffer
+                security,
+                HandleInheritability.None);
+#else
```
Comment (jaredpar) [inline src/Compilers/Server/VBCSCompiler/NamedPipeClientConnection.cs]: add a link to this bug and a brief comment
https://github.com/dotnet/corefx/issues/24040

Comment (khyperia) [reply src/Compilers/Server/VBCSCompiler/NamedPipeClientConnection.cs]: 👍, didn't see that was created already when I updated, so I just wanted to get code out before adding the comment.

**Comment by khyperia** [comment]
retest windows_determinism_prtest please

**Comment by jaredpar** [comment]
> I'm curious if multiline triggers work, because they should, as they have the (?m) flag, but I've been told otherwise...

they do not work

**Comment by khyperia** [comment]
retest windows_determinism_prtest please

**Comment by khyperia** [comment]
retest windows_debug_unit64_prtest please

**Pull request merged**